### PR TITLE
:book: Add netlify configuration

### PR DESCRIPTION
**Pull request by randomvariable**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Part of #1896 

* Refactored the makefiles a fair amount so that the top-level one is cleaner, and there's more code-reuse.
* Added the mdbook bits
* Added the initial site structure

### CODE REVIEW

**Comment by randomvariable** [comment]
/hold

have a slight error in the makefile

**Comment by randomvariable** [comment]
/unhold

I think this is good to go, but I might need to chase up the TLS certificate for cluster-api-aws.sigs.k8s.io

**Comment by randomvariable** [comment]
Modifed the makefile substantially, so just wanna check it all works.

**Comment by randomvariable** [comment]
make targets do appear to work correctly.

rebased.

File: Makefile [hunk 1]
```diff
@@ -12,29 +12,19 @@
...
-# Active module mode, as we use go modules to manage dependencies
-export GO111MODULE=on
+# If you update this file, please follow
+# https://suva.sh/posts/well-documented-makefiles
 
 # Directories.
 ARTIFACTS ?= $(REPO_ROOT)/_artifacts
 TOOLS_DIR := hack/tools
+TOOLS_DIR_DEPS := $(TOOLS_DIR)/go.sum $(TOOLS_DIR)/go.mod $(TOOLS_DIR)/Makefile
 TOOLS_BIN_DIR := $(TOOLS_DIR)/bin
```
Comment (ncdc) [inline Makefile:26]: These are now in common.mk - do you need them here too?

File: docs/book/src/introduction.md [hunk 2]
```diff
@@ -0,0 +1 @@
+{{#include ../../../README.md:2:}}
```
Comment (ncdc) [inline docs/book/src/introduction.md:1]: What does the 2 mean?

**Review by ncdc — COMMENTED** [review]
Just a couple of minor questions/comments but nothing you have to change before getting this merged if you don't want to. Thanks for doing this!!!

File: docs/book/src/introduction.md (hunk 2 above)
Comment (randomvariable) [reply docs/book/src/introduction.md:1]: Ignore the first two lines, so we don't get

```
# Cluster API

# Cluster API
```
on the front page.

**Pull request merged**